packages/assets-controllers/src/crypto-compare-service/crypto-compare.ts

```diff
@@ -0,0 +1,154 @@
+import { handleFetch } from '@metamask/controller-utils';
+
+/**
+ * A map from native currency symbol to CryptoCompare identifier.
+ * This is only needed when the values don't match.
+ */
+const nativeSymbolOverrides = new Map([['MNT', 'MANTLE']]);
+
+const CRYPTO_COMPARE_DOMAIN = 'https://min-api.cryptocompare.com';
+
+/**
+ * Get the CryptoCompare API URL for getting the conversion rate from the given native currency to
+ * the given currency. Optionally, the conversion rate from the native currency to USD can also be
+ * included in the response.
+ *
+ * @param currentCurrency - The currency to get a conversion rate for.
+ * @param nativeCurrency - The native currency to convert from.
+ * @param includeUSDRate - Whether or not the native currency to USD conversion rate should be
+ * included in the response as well.
+ * @returns The API URL for getting the conversion rate.
+ */
+function getPricingURL(
+  currentCurrency: string,
+  nativeCurrency: string,
+  includeUSDRate?: boolean,
+) {
+  nativeCurrency = nativeCurrency.toUpperCase();
+  const fsym = nativeSymbolOverrides.get(nativeCurrency) ?? nativeCurrency;
+  return (
```

Comment:
Should we also use `URLSearchParams` here too?

Reply:
Yes, but this method was already implemented so I rather do this change outside of this PR